packages/opentelemetry-scope-base/src/types.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface BaseScopeManager {
+  /**
+   * Get the current active scope
+   */
+  active(): unknown;
+
+  /**
+   * Run the fn callback with object set as the current active scope
+   * @param scope Any object to set as the current active scope
+   * @param fn A callback to be imediately run within a specific scope
+   */
+  with(scope: unknown, fn: Function): void;
```

Comment:
Should this return the return type of the function that is called?

Reply:
is the coupling between https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-types/src/trace/tracer.ts#L72

```
// in tracer: 
withSpan(span, fn) {
  return this.scopeManager.with(span, function() {
    return fn();
 });
}
```

// in userland:
```
server.use(function(req, res, next) {
     const span = tracer.starr('server');
     tracer.withSpan(span, function() {
       next();
     });
});
```

or do I miss something?

Reply:
@draffensperger Yes, i've updated it.
@hekike Yes, you could however simplify with :
```
withSpan(span, fn) {
  return this.scopeManager.with(span, fn);
}
```
and
```
server.use(function(req, res, next) {
  const span = tracer.start('server');
  tracer.withSpan(span, next);
});
```